File: backend/api/v1/interview.py
from fastapi import APIRouter

from backend.schemas.interview import InterviewRequest
from backend.use_cases.interview_use_case import InterviewUseCase

router = APIRouter(
    prefix="/interview",
    tags=["Interview"],
)


@router.post("/generate")
async def generate_questions(
    request: InterviewRequest,
):

    use_case = InterviewUseCase()

    return use_case.execute(
        request.resume_data,
        request.target_role,
        request.job_description,
    )

File: web/src/services/interview.ts
import api from "./api";

export async function generateInterview(

    resumeData:any,
    targetRole:string,
    jobDescription:string

){

    const response = await api.post(

        "/interview/generate",

        {

            resume_data:resumeData,

            target_role:targetRole,

            job_description:jobDescription

        }

    );

    return response.data;

}